Симонов Кирилл, CSC, hw #1

seemann9 opened the pull request.

seemann9 committed:
initial commit
hw #1 ready

seemann9 committed:
Ready

EduardovEduard commented:
Добрый день, Кирилл.

Вы реализовали сервер одного клиента, то есть сильно облегчили себе задачу в плане синхронизации. Конечно, предполагалось, что сервер может обслуживать несколько клиентов одновременно.

Есть еще несколько вопросов по коду:
Поясните необходимость синхронизации на byId в 77 строке?
Вы не обрабатываете ситуацию зависимости от несуществующей задачи при посылке submit.
в 155 нет реализации шаблона while для wait. Это может привести к преждевременному пробуждению ждущего потока из за spurious wakeup.
Запись задачи в byId делается раньше чем необходимо. Если задача поставленна некорректно, например переменная зависит, от задачи которой нет, такую задачу не стоит записывать и нужно отправить клиенту статус Error

kinmanz commented:
Здравствуйте. Я Сергей. Вы нечаянно отправили это письмо не мне.)

09.05.2016, 18:41, "Эдуардов Эдуард" notifications@github.com:

> Добрый день, Кирилл.
> 
> Вы реализовали сервер одного клиента, то есть сильно облегчили себе задачу в плане синхронизации. Конечно, предполагалось, что сервер может обслуживать несколько клиентов одновременно.
> 
> Есть еще несколько вопросов по коду:
> Поясните необходимость синхронизации на byId в 77 строке?
> Вы не обрабатываете ситуацию зависимости от несуществующей задачи при посылке submit.
> в 155 нет реализации шаблона while для wait. Это может привести к преждевременному пробуждению ждущего потока из за spurious wakeup.
> Запись задачи в byId делается раньше чем необходимо. Если задача поставленна некорректно, например переменная зависит, от задачи которой нет, такую задачу не стоит записывать и нужно отправить клиенту статус Error
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub

kinmanz commented:
не тому кому надо*

09.05.2016, 18:41, "Эдуардов Эдуард" notifications@github.com:

> Добрый день, Кирилл.
> 
> Вы реализовали сервер одного клиента, то есть сильно облегчили себе задачу в плане синхронизации. Конечно, предполагалось, что сервер может обслуживать несколько клиентов одновременно.
> 
> Есть еще несколько вопросов по коду:
> Поясните необходимость синхронизации на byId в 77 строке?
> Вы не обрабатываете ситуацию зависимости от несуществующей задачи при посылке submit.
> в 155 нет реализации шаблона while для wait. Это может привести к преждевременному пробуждению ждущего потока из за spurious wakeup.
> Запись задачи в byId делается раньше чем необходимо. Если задача поставленна некорректно, например переменная зависит, от задачи которой нет, такую задачу не стоит записывать и нужно отправить клиенту статус Error
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub

seemann9 committed:
New features:
Can serve multiple clients simultaneously.
Incorrect task ID in request results in ERROR.

Bugs fixed:
One spurious wakeup issue.

seemann9 commented:
Fixed.
